src/bpfprogram.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "mapmanager.h"
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+namespace bpftrace {
+
+using BpfBytecode = std::unordered_map<std::string, std::vector<uint8_t>>;
```

Comment:
naming nit: BpfBytecodeMap?

Reply:
IMHO no, it's not a map of "multiple BPF bytecodes", the entire map represents a "single BPF bytecode" as it's a map of its sections to the section contents.